boto3/resources/factory.py

```diff
@@ -194,6 +194,17 @@ def _load_waiters(self, attrs, model):
         for waiter in model.waiters:
             attrs[waiter.name] = self._create_waiter(waiter)
 
+    def _create_identifier(factory_self, name):
+        """
+        Creates a read-only property for identifier attributes.
+        """
+        def identifier(self):
+            return getattr(self, '_' + name, None)
```

Comment:
Scratch what I said, I remembered why I set it a default to ``None``. It is because when the resource is instantiated there is check to make sure that all of the identifiers are non-``None`` values: https://github.com/boto/boto3/blob/develop/boto3/resources/base.py#L117. So if I did not use a default, I would get a bad error message like: ``'test.Queue' object has no attribute '_url'`` instead of the current message of: ``ValueError: Required parameter url not set`` because the identifier getter is called before the ValueError is thrown. I would prefer to keep it as is unless you have any other suggestions.